tests/models/test_models.py

```diff
@@ -137,28 +159,43 @@ def test_translate_replace(
     ) -> None:
         self._test_translate(org, driver, test_case_path, "replace")
 
-    def _test_merge(
-        self, org: str, driver: str, test_case_path: pathlib.Path, mode: str
-    ) -> None:
-        with open(test_case_path.joinpath("data_candidate.json"), "r") as f:
-            candidate = json.load(f)
-        with open(test_case_path.joinpath("data_running.json"), "r") as f:
-            running = json.load(f)
-        with open(test_case_path.joinpath(f"res_{mode}"), "r") as f:
-            expected = f.read()
-        driver_class = ntc_rosetta.get_driver(driver, org)
-        device = driver_class()
-        res_merge = device.merge(candidate, running, replace=mode == "replace")
-        assert res_merge == expected
-
-    @pytest.mark.parametrize(**get_test_cases("merge"))  # type: ignore
-    def test_merge_merge(
-        self, org: str, model: str, driver: str, test_case_path: pathlib.Path
-    ) -> None:
-        self._test_merge(org, driver, test_case_path, "merge")
 
-    @pytest.mark.parametrize(**get_test_cases("merge"))  # type: ignore
-    def test_merge_replace(
-        self, org: str, model: str, driver: str, test_case_path: pathlib.Path
-    ) -> None:
-        self._test_merge(org, driver, test_case_path, "replace")
+def parse_path(pathname: str) -> Data:
+    """Parse a given test path into a useful object
+
+    Args:
+        pathname: string, path must conform to
+            `tests/models/{org}/data/{model}/{action}/{driver}`
+
+    Returns:
+        A namedtuple of strings
+
+    """
+    path = pathlib.Path(pathname)
+    action = path.parents[2].stem
+    driver = path.parents[1].stem
+    model = path.parents[3].stem.replace("_", "-")
+    org = path.parents[5].stem
+    return Data(org, model, driver, action, path.parent)
+
+
+def merge(org: str, driver: str, test_case_path: pathlib.Path, mode: str) -> None:
```

Comment:
I don't really see another solution for the testing to be able to be more granular than the way you've refactored to.

I would say you can go ahead and do the same for the translate/parse as well so they're all structured the same way.

The way this is refactored should also be able to address #3 if we go down the road of raising an exception for parsing failures.